tools/autograd/load_derivatives.py

```diff
@@ -19,18 +19,30 @@ def load_derivatives(path, declarations):
     for declaration in declarations:
         declarations_by_signature[get_signature(declaration)].append(declaration)
 
-    autograd_functions = [
+    differentiability_infos = [
         process_definition(defn, declarations_by_signature)
         for defn in definitions]
+
+    autograd_functions = [d['autograd_fn'] for d in differentiability_infos if d['autograd_fn'] is not None]
     ensure_unique_names(autograd_functions)
-    match_declarations_with_autograd_functions(declarations, autograd_functions)
+    match_declarations_with_differentiability_info(declarations, differentiability_infos)
 
     return autograd_functions
 
 
+def create_differentiability_info(signature, output_differentiability,
+                                  autograd_fn):
+    return {
+        'signature': signature,
+        'output_differentiability': output_differentiability,
+        'autograd_fn': autograd_fn,
+    }
+
+
 # How do you feel about pasting declaration inside autograd function...
-def create_autograd_function(name, derivatives, args_with_derivatives, non_differentiable_arg_names,
-                             signature, declaration, output_differentiability):
+def create_autograd_function(name, derivatives, args_with_derivatives,
+                             declaration, output_differentiability,
```

Comment:
Is the 'output_differentiability' argument still needed?

Reply:
No, good catch.